Add mcp server exposing wolfram alpha llm api

mariofusco opened the pull request.

maxandersen commented on containers/pom.xml:
```diff
@@ -34,6 +34,18 @@
             <artifactId>mcp-server-shared</artifactId>
             <version>${project.version}</version>
         </dependency>
+
```
why adding all these ? they are already in parent pom?

maxandersen replied:
these changes to child submodules aren't needed afaics.

maxandersen commented on wolframalpha-llm/README.md:
````diff
@@ -0,0 +1,37 @@
+
+Use the following command to start the server.
+```shell
+jjbang --quiet -DWOLFRAM_LLM_API_KEY=${YOUR_WOLFRAM_APP_ID} io.quarkiverse.mcp.servers:mcp-server-wolframalpha-llm:999-SNAPSHOT:runner
````
```suggestion
jbang -DWOLFRAM_LLM_API_KEY=${YOUR_WOLFRAM_APP_ID} io.quarkiverse.mcp.servers:mcp-server-wolframalpha-llm:RELEASE:runner
```

minor fixes - but you can just add it to jbang-catalog and you can run it using that.

maxandersen replied:
p.s. when I do mvn clean install no main class generated ...debugging why :)

maxandersen commented on wolframalpha-llm/src/main/java/org/acme/WebSearchService.java:
```diff
@@ -0,0 +1,39 @@
+package org.acme;
```
should use the quarkiverse package names  -I reckon it came from template? should fix that :)

maxandersen replied:
no, template actually has it correct :) `jbang init -t mcp wolframalpha` would have done it "right"

maxandersen commented on wolframalpha-llm/src/main/java/org/acme/WebSearchService.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    private static String doSearch(String webUrl) throws IOException {
+        String text = Jsoup.connect(webUrl).get().text();
```
is it really html being returned? the docs state its easily consumable text?

maxandersen commented on wolframalpha-llm/README.md:
```diff
@@ -0,0 +1,37 @@
+
+This Model Context Protocol(MCP) server enables Large Language Models (LLMs) to perform web searches on the [Wolfram Alpha LLM API](https://products.wolframalpha.com/llm-api/documentation).
+
+## General Usage
```
for all the others we got a screenshot or usecase shown - got something interesting we can put in here?

Also, would be great adding notice to the root readme.

maxandersen commented on wolframalpha-llm/pom.xml:
```diff
@@ -0,0 +1,121 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>mcp-server-wolframalpha-llm</artifactId>
```
could just call it wolframalpha ? makes it shorter/easier to type/use.

mariofusco replied:
Wolfram Alpha exposes many different APIs and the one that I'm using is specific to be used with a LLM, so probably it's better to also keep that suffix in case we also want to integrate other APIs.

maxandersen replied:
wouldn't we just add that to an wolfram mcp rather than split it?

maxandersen commented on wolframalpha-llm/src/main/resources/application.properties:
```diff
@@ -0,0 +1,6 @@
+quarkus.package.jar.type=uber-jar
```
this is unnecesary as its set in shared.

mariofusco replied:
I'm not depending on shared, should I?

maxandersen replied:
yes please - gives you ability to turn on/off sse

maxandersen commented on wolframalpha-llm/pom.xml:
```diff
@@ -0,0 +1,121 @@
+    </dependencyManagement>
+
+    <dependencies>
+        <dependency>
```
should add dependency to shared module so you get the config/bootstrap logic automatically provided.

mariofusco replied:
Ok, let me try.

mariofusco commented:
@maxandersen Thanks for review. I addressed all your comments, except the one on adding a screenshot: this is simply performing a web search so I'm afraid there is no much to show here. I will demonstrate how this works when I will integrate it with my agentic AI examples.

Please review again.

maxandersen commented on wolfram/README.md:
````diff
@@ -0,0 +1,37 @@
+To get more detailed logging you can add the following parameters to the jbang command line:
+
+```shell
+-Dquarkus.log.file.enable=true -Dquarkus.log.file.path=${user.home}/mcp-server-filesystem.log
````
should refer to wolfram
```suggestion
-Dquarkus.log.file.enable=true -Dquarkus.log.file.path=${user.home}/mcp-server-wolfram.log
```

maxandersen commented on wolfram/src/main/resources/application.properties:
```diff
@@ -0,0 +1,4 @@
+quarkus.log.file.enable=true
```
as mcp clients normally log output normally this is redundant. users can enable it themself if need be.

Also since it generates in working directory the file is going to show up wherever mcp client runs it rather in your own directory. 

Thus lets remove it.

maxandersen reviewed (changes requested):
added a few comments as I think its still changing more than it needs.

mariofusco commented:
> added a few comments as I think its still changing more than it needs.

I should have addressed all of them, review again, thanks.

mariofusco commented:
@maxandersen could this be merged or is there anything still pending here?

maxandersen commented:
@mariofusco the question about why changing all the pom.xml files is still needed?

mariofusco commented:
> @mariofusco the question about why changing all the pom.xml files is still needed?

No, this is no longer necessary to make my server to work, so I could revert it, but if I understand correctly you wanted to do this anyway, see https://github.com/quarkiverse/quarkus-mcp-servers/blob/59ceb21aa28c4c6025b66ef639893ca8a514e59a/pom.xml#L76

@maxandersen Let me know if I should revert the changes to all poms.

maxandersen commented:
yeah - just revert those for now. the todo was there when we couldn't easily turn it on/off/combine them. we can now.

mariofusco commented:
> yeah - just revert those for now. the todo was there when we couldn't easily turn it on/off/combine them. we can now.

Ok, done.

maxandersen reviewed (approved):
lgtm
